apps/cloud_inference/server_app/source/inference_hip_rocal.cpp

```diff
@@ -0,0 +1,1092 @@
+            fatal("workDeviceInputCopy: unexpected nullptr in queueDeviceInputMemIdle[%d] mapped_ptr", gpu);
+        }
+        int rem_images = rocalGetRemainingImages(rocalHandle[gpu]);
+        int inputCount = rem_images < batchSize ? rem_images : batchSize;
+
+        // decode and resize using rocAL
+        if(rocalRun(rocalHandle[gpu]) != 0) {
+            printf("workDeviceInputCopy: rocalRun() failed for gpu : [%d]", gpu);
+            break;
+        }
+        
+        RocalTensorOutputType tensor_output_type = useFp16 ? RocalTensorOutputType::ROCAL_FP16 : RocalTensorOutputType::ROCAL_FP32;
+        RocalTensorLayout tensor_format = RocalTensorLayout::ROCAL_NCHW;
+        
+        if(rocalCopyToOutputTensor(rocalHandle[gpu], mapped_ptr, tensor_format, tensor_output_type, preprocessMpy[0],
+            preprocessMpy[1], preprocessMpy[2], preprocessAdd[0], preprocessAdd[1], preprocessAdd[2], reverseInputChannelOrder) != 0) {
+            printf("workDeviceInputCopy: rocalCopyToOutputTensor() failed for gpu : [%d]", gpu);
+        }
+        
+        std::vector<std::string> names;
+        names.resize(inputCount);
+
+        int image_name_length[inputCount];
+        int img_name_size = rocalGetImageNameLen(rocalHandle[gpu], image_name_length);
+        char img_names[img_name_size];
+        rocalGetImageName(rocalHandle[gpu], img_names);
+        std::string imageNamesStr(img_names);
+
+        int pos = 0;
+        for(int i = 0; i < inputCount; i++) {
+            names[i] = imageNamesStr.substr(pos, image_name_length[i]);
+            pos += image_name_length[i];
+            
+            queueDeviceNameQ[gpu]->enqueue(names[i]);
+        }
+        
+        if(rocalIsEmpty(rocalHandle[gpu])) {
+            if(!loop) {
+                endOfSequenceReached = true;
+            }
+        }
+
+        if(hipStreamSynchronize(stream) != hipSuccess) {
+            fatal("workDeviceInputCopy: hipStreamSynchronize(#%d) failed", gpu);
+        }
+
+        if(inputCount > 0) {
+            // add the input for processing
+            queueDeviceInputMemBusy[gpu]->enqueue(input);
+            // update counters
+            totalBatchCounter++;
+            totalImageCounter += inputCount;
+        }
+        else {
+            // add the input back to idle queue
+            queueDeviceInputMemIdle[gpu]->enqueue(input);
+        }
+        PROFILER_STOP(inference_server_app, workDeviceInputCopyBatch);
+    }
+    // release HIP stream
+    hipStreamDestroy(stream);
+
+    // add the endOfSequenceMarker to next stage
+    void* endOfSequenceMarker = nullptr;
+    queueDeviceInputMemBusy[gpu]->enqueue(std::make_pair(endOfSequenceMarker, endOfSequenceMarker));
+
+    args->lock();
+    info("workDeviceInputCopy: GPU#%d terminated for %s [processed %d batches, %d images]", gpu, clientName.c_str(), totalBatchCounter, totalImageCounter);
+    args->unlock();
+}
+
+void InferenceEngineRocalHip::workDeviceProcess(int gpu)
```

Comment:
if this function is same as base class, do not redefine here

Reply:
Removed